weldx/asdf/util.py

```diff
@@ -542,6 +553,80 @@ def _get_instance_shape(
     return None
 
 
+class _ProtectedViewDict(MutableMapping):
+    def __init__(self, protected_keys, data=None):
+        super(_ProtectedViewDict, self).__init__()
+        self._data = data
+        self.protected_keys = protected_keys
+
+    def __len__(self) -> int:
+        return len(self.keys())
+
+    def __getitem__(self, key):
+        if key in self.protected_keys:
+            self._warn_protected_keys()
+            raise KeyError
+        return self._data.get(key)
+
+    def __delitem__(self, key):
+        if key in self.protected_keys:
+            self._warn_protected_keys()
+            return
+        del self._data[key]
+
+    def __setitem__(self, key, value):
+        if key in self.protected_keys:
+            self._warn_protected_keys()
+            return
+        self._data[key] = value
+
+    def keys(self) -> AbstractSet:
+        return {k for k in self._data.keys() if k not in self.protected_keys}
+
+    def __iter__(self):
+        return (k for k in self.keys())
+
+    def __contains__(self, item):
+        return item in self.keys()
+
+    def update(
+        self, mapping: Mapping[Hashable, Any], **kwargs: Any
+    ):  # pylint: disable=W0221
+        _mapping = dict(mapping, **kwargs)  # merge mapping and kwargs
+        if any(key in self.protected_keys for key in _mapping.keys()):
+            self._warn_protected_keys()
+            _mapping = {
+                k: v for k, v in _mapping.items() if k not in self.protected_keys
+            }
+
+        self._data.update(_mapping)
+
+    def popitem(self) -> Tuple[Hashable, Any]:
+        for k in self.keys():
+            if k in self.protected_keys:
+                continue
+
+            return k, self.pop(k)
```

Comment:
```suggestion
        for k in self.keys():
            if k not in self.protected_keys:
                return k, self.pop(k)
```

Reply:
I thought I write it in a more readable manner (as you could quickly overlook the negation in the condition). I guess it is a matter of taste, not runtime.

Reply:
Jep, shouldn't have any effect on runtime. So your decision :wink